src/internal/TLC5947GenericMethod.h

```diff
@@ -0,0 +1,253 @@
+/*-------------------------------------------------------------------------
+NeoPixel library helper functions for TLC5947 24 channel PWM controller using general Pins.
+
+Written by Michael C. Miller.
+Written by Dennis Kasprzyk.
+
+I invest time and resources providing this open source code,
+please support me by dontating (see https://github.com/Makuna/NeoPixelBus)
+
+-------------------------------------------------------------------------
+This file is part of the Makuna/NeoPixelBus library.
+
+NeoPixelBus is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, either version 3 of
+the License, or (at your option) any later version.
+
+NeoPixelBus is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with NeoPixel.  If not, see
+<http://www.gnu.org/licenses/>.
+-------------------------------------------------------------------------*/
+
+#pragma once
+
+// must also check for arm due to Teensy incorrectly having ARDUINO_ARCH_AVR set
+#if defined(ARDUINO_ARCH_AVR) && !defined(__arm__)
+#include "TwoWireBitBangImpleAvr.h"
+#else
+#include "TwoWireBitBangImple.h"
+#endif
+
+class TLC5947ConverterBase
+{
+public:
+};
+
+class TLC5947Converter8Bit : TLC5947ConverterBase
```

Comment:
`TLC5947Converter8Bit` does not follow code guidelines.  Object names are only capitalized for the first letter of each "word" part.  An acronym or even a product name is considered a word and only the first letter should be capitalized.
`Tlc5947Converter8Bit`
There are others below, I will only comment once here and expect you to apply the concept across the code change.